src/test/java/com/mongodb/hibernate/jdbc/MongoResultSetTests.java

```diff
@@ -0,0 +1,282 @@
+        void testColumnIndexUnderflow(String label, ResultSetMethodInvocation methodInvocation) {
+            var exception = assertThrows(SQLException.class, () -> methodInvocation.runOn(mongoResultSet));
+            assertTrue(exception.getMessage().startsWith("Invalid column index"));
+        }
+
+        @ParameterizedTest(
+                name = "SQLException is thrown when \"{0}\" is called on a MongoResultSet with columnIndex too high")
+        @MethodSource("getMongoResultSetMethodInvocationsWithColumnIndexOverflow")
+        void testColumnIndexOverflow(String label, ResultSetMethodInvocation methodInvocation) {
+            var exception = assertThrows(SQLException.class, () -> methodInvocation.runOn(mongoResultSet));
+            assertTrue(exception.getMessage().startsWith("Invalid column index"));
+        }
+
+        private static Stream<Arguments> getMongoResultSetMethodInvocationsWithColumnIndexUnderflow() {
+            return doGetMongoResultSetMethodInvocationsImpactedByColumnIndex(0);
+        }
+
+        private static Stream<Arguments> getMongoResultSetMethodInvocationsWithColumnIndexOverflow() {
+            return doGetMongoResultSetMethodInvocationsImpactedByColumnIndex(FIELDS.size() + 1);
+        }
+    }
+
+    static Stream<Arguments> doGetMongoResultSetMethodInvocationsImpactedByClosing() {
+        var invocationsImpactedByColumnIndex = doGetMongoResultSetMethodInvocationsImpactedByColumnIndex(1);
+        var additionalArguments = Map.<String, ResultSetMethodInvocation>ofEntries(
+                        Map.entry("next()", ResultSet::next),
+                        Map.entry("wasNull()", ResultSet::wasNull),
+                        Map.entry("getMetaData()", ResultSet::getMetaData),
+                        Map.entry("findColumn(String)", rs -> rs.findColumn("name")),
+                        Map.entry("isWrapperFor(Class)", rs -> rs.isWrapperFor(MongoResultSet.class)))
+                .entrySet()
+                .stream()
+                .map(entry -> Arguments.of(entry.getKey(), entry.getValue()));
+        return Stream.concat(invocationsImpactedByColumnIndex, additionalArguments);
+    }
+
+    static Stream<Arguments> doGetMongoResultSetMethodInvocationsImpactedByColumnIndex(int columnIndex) {
+        return Map.<String, ResultSetMethodInvocation>ofEntries(
+                        Map.entry("getString(int)", rs -> rs.getString(columnIndex)),
+                        Map.entry("getBoolean(int)", rs -> rs.getBoolean(columnIndex)),
+                        Map.entry("getByte(int)", rs -> rs.getByte(columnIndex)),
+                        Map.entry("getShort(int)", rs -> rs.getShort(columnIndex)),
+                        Map.entry("getInt(int)", rs -> rs.getInt(columnIndex)),
+                        Map.entry("getLong(int)", rs -> rs.getLong(columnIndex)),
+                        Map.entry("getFloat(int)", rs -> rs.getFloat(columnIndex)),
```

Comment:
Should we be testing that named params throw exceptions? (I do not have a view on this, just want to confirm.)

Reply:
I doubt the necessity to test adapter. If we didn't implement API methods, we could assume exception would be thrown